open-apparel-registry/open-apparel-registry
# Enable users to upload and view facility lists

## Overview

This PR enables users to upload and view facility lists through the UI.

Closes #101 
Connects #148
Connects #105 

## Demo

![facility-lists](https://user-images.githubusercontent.com/4165523/51869124-ab82a380-231d-11e9-878f-9a2f9f0adbed.gif)
![facility-lists-2](https://user-images.githubusercontent.com/4165523/51869125-ab82a380-231d-11e9-86a8-6e9ea0e69c11.gif)

## Testing Instructions

- get this branch then
- `./scripts/manage reset_schema`
- `./scripts/manage migrate`
- `./scripts/server`
- visit the app on 6543 and register as a new user
- log in
- click on "Contribute" and verify that you are taken to the contribute form
- verify that you are able to upload new lists in the following series of conditions:
    - name and description empty
    - name or description empty
    - name and description but no replaces
    - name, description, and replaces
- verify that a list that has been replaced is listed as "(inactive)" in the dropdown
- click the link to view your lists
- verify that you see the table above
- sign out
- visit :6543/lists and verify that you have an indication that you should sign in
- visit :6543/contribute and verify that you have an indication that you should sign in
